internal/httputil/handlers.go

```diff
@@ -27,6 +28,20 @@ func Redirect(w http.ResponseWriter, r *http.Request, url string, code int) {
 	http.Redirect(w, r, url, code)
 }
 
+// RenderJSON replies to the request with the specified struct as JSON and HTTP code.
+// It does not otherwise end the request; the caller should ensure no further
+// writes are done to w.
+// The error message should be application/json.
+func RenderJSON(w http.ResponseWriter, code int, v interface{}) {
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(code)
+	if err := json.NewEncoder(w).Encode(v); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
we likely need to encode the JSON to a buffer in order to be able to set the header like this. Otherwise we will have already written the headers when the error occurs.

Reply:
Good catch (... and... shame on me because we definitely discussed this _exact_ issue last week !). 

I updated the code using a bytes buffer and added a test to check the header value. Let me know what you think.